src/main/docs/serializers.adoc

```diff
@@ -0,0 +1,265 @@
+But using the empty state the serializer will not able to map that value to a fruit name, and so it would throw.
+That leaves the persisted data unusable.
+So what could you have done differently to make it work? The answer is simple. Persist the serializer's state as well
+and restore it when the cache is re-initialized.
+And that is exactly what persistent serializers would do.
+
+=== Persistent serializers
+
+A persistent serializer persists its state and retrieves it when reinitialized.
+A persistent serializer implementation can choose to persist the data wherever it wants.
+But a recommended way is to use the cache manager's `LocalPersistenceService` so that the cache manager would take care
+of the persistence.
+Inorder to do that, the serializer implementation needs to have a constructor that takes in a
+`FileBasedPersistenceContext` as an argument, in addition to class loader argument.
+The use of the `FileBasedPersistenceContext` argument is optional.
+But the presence of this double-argument constructor is a strict requirement for persistent caches.
+When the cache using this serializer is initialized, this 2-argument constructor is used to instantiate the serializer.
+Have a look at this dummy implementation of a persistent serializer.
+It is just an extension of the same old transient serializer with the persistent stuff wired in.
+
+[source,java,indent=0]
+----
+include::{sourcedir}/test/java/org/ehcache/demo/serializer/DumbPersistentStringSerializer.java[tag=persistentSerializer]
+----
+
+In the above persistent serializer, the state or the serialization/deserialization has not changed.
+The only additional thing is the persistence logic. And that is fairly simple too.
+The state is restored on initialization if one is found, and persisted on close.
+And have a look at the sample from the previous section modified to use our persistent serializer.
+
+[source,java,indent=0]
+----
+include::{sourcedir}/test/java/org/ehcache/demo/serializer/SerializersDemo.java[tag=persistentSerializerGoodSample]
+----
+
+<1> The only change from the previous sample is the usage of `DumbPersistentStringSerializer` here.
+
+== Third-party serializers
+
+Ehcache by-default relies on a tweaked form of java standard serialization to perform serialization and deserialization.
+But most of you already know that java built-in serialization is not the best performing serialization technique.
+A lot of alternative serialization techniques are available in the market.
+With the custom serializers support of ehcache you can take advantage of any one of those third-party serializers out 
+there and use those within ehcache.
+All you have to do is write a custom serializer using the third-party serializer of your choice.
+
+In-order to demonstrate that, I have written a custom serializer using the popular serialization framework *Kryo*.
+There is no specific reason for me to pick *kryo* in my example.
```

Comment:
I would drop this sentence and the following ones. Either list alternatives or keep your privilege as a writer for making choices, no need to justify them.